metadata_scripts/GCEMetadataScripts/main.go

```diff
@@ -40,130 +43,45 @@ import (
 
 var (
 	programName    = "GCEMetadataScripts"
+	version        = "dev"
 	metadataURL    = "http://metadata.google.internal/computeMetadata/v1"
 	metadataHang   = "/?recursive=true&alt=json&timeout_sec=10&last_etag=NONE"
 	defaultTimeout = 20 * time.Second
-	commands       = []string{"specialize", "startup", "shutdown"}
-	scripts        = map[metadataScriptType]string{
-		ps1: "%s-script-ps1",
-		cmd: "%s-script-cmd",
-		bat: "%s-script-bat",
-		uri: "%s-script-url",
-	}
-	version        string
 	powerShellArgs = []string{"-NoProfile", "-NoLogo", "-ExecutionPolicy", "Unrestricted", "-File"}
+	usageError     = fmt.Errorf("No valid arguments specified. Specify one of \"startup\", \"shutdown\" or \"specialize\"")
 
 	storageURL = "storage.googleapis.com"
 
 	bucket = `([a-z0-9][-_.a-z0-9]*)`
 	object = `(.+)`
+
 	// Many of the Google Storage URLs are supported below.
 	// It is preferred that customers specify their object using
 	// its gs://<bucket>/<object> URL.
 	bucketRegex = regexp.MustCompile(fmt.Sprintf(`^gs://%s/?$`, bucket))
 	gsRegex     = regexp.MustCompile(fmt.Sprintf(`^gs://%s/%s$`, bucket, object))
+
 	// Check for the Google Storage URLs:
 	// http://<bucket>.storage.googleapis.com/<object>
 	// https://<bucket>.storage.googleapis.com/<object>
 	gsHTTPRegex1 = regexp.MustCompile(fmt.Sprintf(`^http[s]?://%s\.storage\.googleapis\.com/%s$`, bucket, object))
+
 	// http://storage.cloud.google.com/<bucket>/<object>
 	// https://storage.cloud.google.com/<bucket>/<object>
 	gsHTTPRegex2 = regexp.MustCompile(fmt.Sprintf(`^http[s]?://storage\.cloud\.google\.com/%s/%s$`, bucket, object))
+
 	// Check for the other possible Google Storage URLs:
 	// http://storage.googleapis.com/<bucket>/<object>
 	// https://storage.googleapis.com/<bucket>/<object>
 	//
-	// The following are deprecated but checked:
+	// The following are deprecated but also checked:
 	// http://commondatastorage.googleapis.com/<bucket>/<object>
 	// https://commondatastorage.googleapis.com/<bucket>/<object>
 	gsHTTPRegex3 = regexp.MustCompile(fmt.Sprintf(`^http[s]?://(?:commondata)?storage\.googleapis\.com/%s/%s$`, bucket, object))
 
 	testStorageClient *storage.Client
 )
 
-const (
-	ps1 metadataScriptType = iota
-	cmd
-	bat
-	uri
-)
-
-type metadataScriptType int
-
-type metadataScript struct {
-	Type             metadataScriptType
-	Script, Metadata string
-}
-
-func prepareURIExec(ctx context.Context, ms *metadataScript) (*exec.Cmd, string, error) {
-	trimmed := strings.TrimSpace(ms.Script)
-	bucket, object := findMatch(trimmed)
-	if bucket != "" && object != "" {
-		trimmed = fmt.Sprintf("https://%s/%s/%s", storageURL, bucket, object)
```

Comment:
FYI I did not understand why the URL was converted to GCS here but then again in downloadScript(), since the latter is only called from here. So this duplication was dropped.